### ISSUE TITLE
Distribute PyOWM as .deb and .rpm packages

### ISSUE THREAD

**Issue by csparpa** [csparpa/pyowm#98]
In order to make even more straightforward for users to install PyOWM, it would be superb to have it installable via package managers such as `yum` or `rpm`.

Eg:

```
$ sudo apt-get install pyowm
$ yum install pyowm
[etc]
```

A good read is: https://wiki.debian.org/Python/Packaging

**Comment by Diapente** [comment]
There's an Arch PKGBUILD in the AUR under the name python-owm.

**Comment by csparpa** [comment]
Hum... I didn't package the lib using any packaging channel, so someone has already kicked it off.

Nothing wrong with it, of course, as soon as that bundle effectively contains PyOWM 2.5

As regards this task, I was thinking about Debian/Ubuntu packaging as a priority

**Comment by Diapente** [comment]
Yeah that was me - it was more of an FYI. It should be at 2.5.0 now.

**Comment by csparpa** [comment]
Haaa sorry, I didn't get :)
Super!!! Thanks for supporting!

**Comment by csparpa** [comment]
@Diapente could you please submit any kind of high level guideline on how to install PyOWM via AUR packaging? Be it anything, I mean, a single command line or a bullet pointed list or whatever
So that I can update READMe.md with that info

You could submit a PR to the master branch yourself as well, if you want

Thanks so much!

**Comment by Diapente** [comment]
Sure - there are various methods to access the AUR but one of more common/easily understood ones would be:

`Yaourt -S python-owm`

Then follow the prompts as for any other PKGBUILD. 

I only made a python3 package. PyOWM is compatible with python2 so do you want me to package a python2 version? That would take the name python2-owm (following AUR convention).

**Comment by csparpa** [comment]
@Diapente yes please, Python 2 is still extensively used

I added install notes to this new wiki page: if there is anything wrong with it, please notify me!
https://github.com/csparpa/pyowm/wiki/Install

**Comment by Diapente** [comment]
Done - python2-owm is now available from the AUR.

**Comment by csparpa** [comment]
Thanks @Diapente !

**Comment by axodd** [comment]
This is a good idea, as both pyowm and OWM are awesome! Since Wunderground.com is shutting down all of its API access keys (except those of PWS owners) completely starting next month (i.e. after December 31, 2018), many folks will look for alternative web API solutions.
Re .deb, going Debian (rather than Ubuntu) route first is probably the best way to handle it. It ensures that your package will be automatically synced from Debian into Ubuntu, and will be available not only in Ubuntu/Debian, but in all Debian-based derivatives out there as well. Also, it avoids duplication of efforts and conflicting packages across multiple .deb repositories. Here is a nice overview (albeit old) that i stumbled upon:
https://askubuntu.com/questions/16446/how-to-get-my-software-into-ubuntu
Essentially, it comes down to:
 http://packaging.ubuntu.com/html/packaging-new-software.html
 https://wiki.ubuntu.com/UbuntuDevelopment/NewPackages
 https://www.debian.org/doc/manuals/developers-reference/ch05.html#newpackage
 https://wiki.debian.org/Teams/PythonAppsPackagingTeam
 https://salsa.debian.org/python-team/tools/python-apps/blob/master/policy.rst
 https://salsa.debian.org/python-team/applications
 https://tracker.debian.org/teams/python-apps/
As there are thousands Python packages in Ubuntu and Debian repos (prefixed with `python-` for Python 2 and `python3-` for Python 3) that have been accepted by Debian/Ubuntu teams, sure they know that open weather data accessibility/transparency/awareness becomes particularly important in the era of global warming/climate change, so my guess is they could facilitate Pyowm's expedited inclusion! Thanks

**Comment by csparpa** [comment]
@axodd thanks a lot for your effort and reseach!

**Comment by tigerfoot** [comment]
BTW pyowm is available for current (and next) openSUSE release.
The packaging is done here https://build.opensuse.org/package/show/devel:languages:python/python-pyowm
Version python2 and python3 are managed. It's being updated each time we found a new version on pypi

**Comment by csparpa** [comment]
Thanks for notifying @tigerfoot !
